sdk/Questionnaire/Android.mdx

```diff
@@ -0,0 +1,166 @@
+    val createViewHolderDelegate: CreateViewHolderDelegate, // used to create a view holder
+    val pageToViewDataConverter: (
+        FormPage, FormPageToQuestionnaireStepViewDataConverter.LoopFormData?
+    ) -> QuestionnaireStepViewData // converts FormPage to QuestionnaireStepViewData
+)
+
+typealias CreateViewHolderDelegate = (
+    parent: ViewGroup,
+    editQuestionListener: (question: QuestionViewData<*>) -> Unit,
+    answerChangeListener: () -> Unit,
+    launchFlowListener: (LaunchFlowParams) -> Unit,
+    orientation: Int
+) -> QuestionViewHolder
+```
+* **CommonValidator** - defines default questions validation, which is used unless custom `AnswerValidation` is provided (such as validating `lowerBound`, `upperBound` in `NumberQuestionViewData`: 
+
+`fun isValid(question: QuestionViewData<*>): Boolean`
+
+### Preferred units
+
+You can declare a question answer converter which will transform it into a proper `QuestionAnswerViewData` for a default measurement unit, see:  
+```kotlin
+HumaQuestionnaireManager.getInstance().setQuestionAnswerConverter(/*..*/)`
+```
+
+Preferred unit page builder extension function sample:
+
+```kotlin
+fun Pages.height(
+    preferredUnitsContainer: PreferredUnitsContainer,
+    pageId: String,
+    titleStringResId: Int = R.string.onboarding_questionnaire_height,
+    minHeightInCm: Int = MIN_HEIGHT_CM,
+    maxHeightInCm: Int = MAX_HEIGHT_CM
+) {
+    val manager = HumaQuestionnaireManager.getInstance()
+    val unitType = HeightMeasurementUnitType()
+    val defaultUnit = unitType.getDefaultUnit()
+    when (val unit = preferredUnitsContainer.getPreferredUnit(unitType)) {
+        HeightMeasurementUnit.Cm -> {
+            numeric { /*..*/ }
+            manager.setQuestionAnswerConverter(pageId) { answerViewData ->
+                answerViewData as NumberQuestionAnswerViewData
+                val convertedValue = unitType.convert(answerViewData.value, from = unit, to = defaultUnit)
+                NumberQuestionAnswerViewData(convertedValue)
+            }
+        }
+        HeightMeasurementUnit.FtIn -> {
+            twoPickerChoice { /*..*/ }
+            manager.setQuestionAnswerConverter(pageId) { answerViewData ->
+                answerViewData as TwoPickerSelectionQuestionAnswerViewData
+                val feet = FtAndInches.fromPair(answerViewData.value).toFeet()
+                val convertedValue = unitType.convert(feet, from = unit, to = defaultUnit)
+                NumberQuestionAnswerViewData(convertedValue)
+            }
+        }
+    }
+}
+```
+
+</TabItem>
+<TabItem value="installing" label="Installing">
+
+1. Add the dependency in your local build.gradle file:
+```kotlin
+implementation("com.huma.sdk:questionnaire:<latest-version>")
+```
+2. Initialize `HumaQuestionnaireManager` in your Application class:
+```kotlin
+HumaQuestionnaireManager.init(
+   questionnaireSourceRepository: QuestionnaireSourceRepository,
```

Comment:
questionnaireSourceRepository: QuestionnaireSourceRepository is removed from latest sdk